.ci/vfio_jenkins_job_build.sh

```diff
@@ -0,0 +1,294 @@
+#!/bin/bash
+#
+# Copyright (c) 2020 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+# Run the .ci/jenkins_job_build.sh script in a VM
+# that supports VFIO, then run VFIO functional tests
+
+set -x
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o errtrace
+
+cidir=$(dirname "$0")
+
+source /etc/os-release || source /usr/lib/os-release
+source "${cidir}/lib.sh"
+
+http_proxy=${https_proxy:-}
+https_proxy=${https_proxy:-}
+vm_ip="127.0.15.1"
+vm_port="10022"
+data_dir="/tmp/kata-containers/vfio-test"
+ssh_key_file="${data_dir}/key"
+arch=$(uname -i)
+
+mkdir -p "${data_dir}"
+
+trap cleanup EXIT
+
+cleanup() {
+	sudo killall -9 qemu-system-${arch}
+}
+
+create_ssh_key() {
+	rm -f "${ssh_key_file}"
+	ssh-keygen -f "${ssh_key_file}" -t rsa -N ""
+}
+
+create_meta_data() {
+	file="$1"
+	cat <<EOF > "${file}"
+{
+  "uuid": "d1b4aafa-5d75-4f9c-87eb-2ceabe110c39",
+  "hostname": "test"
+}
+EOF
+}
+
+create_user_data() {
+	file="$1"
+	ssh_pub_key_file="$2"
+
+	ssh_pub_key="$(cat "${ssh_pub_key_file}")"
+	dnf_proxy=""
+	environment="$(env | egrep "ghprb|WORK|KATA|GIT|JENKINS" | sed -e 's|=|="|g' -e 's|$|"|g' -e 's|^|    export |g')"
+
+	if [ -n "${http_proxy}" ] && [ -n "${https_proxy}" ]; then
+		dnf_proxy="dnf:
+  https_proxy: ${https_proxy}
+  proxy: ${http_proxy}"
+		if [ -n "${no_proxy}" ]; then
+			environment+="
+    export no_proxy=\"${no_proxy}\"
+    export NO_PROXY=\"${no_proxy}\"
+    export http_proxy=\"${http_proxy}\"
+    export HTTP_PROXY=\"${http_proxy}\"
+    export https_proxy=\"${https_proxy}\"
+    export HTTPS_PROXY=\"${https_proxy}\""
+		fi
+	fi
```

Comment:
@devimc is this formatting off or is like this because you are adding it to a file?

Reply:
it's because I'm adding this to a yaml file, so the format is important